tests/commands/test_update.py

```diff
@@ -32,103 +31,447 @@ def test_update_run(tmp_library: TemporaryLibrary) -> None:
+    assert not result.output
+
+    result = cli_runner.invoke(
+        cli,
+        ["--batch"]
+        + ["--append", "year", "123"]  # this fails
+        + ["--remove", "year", "123"]  # this fails
+        + ["--append", "tags", "tag3"]  # but this should still work
+        + ["krishnamurti"],
+    )
+    assert result.exit_code == 0
+
+    db = papis.database.get()
+    (doc,) = db.query_dict({"author": "Krishnamurti"})
+    assert "tag3" in doc["tags"]
+
+
+def test_update_yaml_cli(
+    tmp_library: TemporaryLibrary,
+    resource_cache: ResourceCache,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    from papis.commands.update import cli
+
     cli_runner = PapisRunner()
 
     filename = os.path.join(resource_cache.cachedir, "update", "russell.yaml")
 
     import papis.utils
+
     with monkeypatch.context() as m:
-        m.setattr(papis.utils, "update_doc_from_data_interactively",
-                  update_doc_from_data_interactively)
+        m.setattr(
+            papis.utils,
+            "update_doc_from_data_interactively",
+            update_doc_from_data_interactively,
+        )
 
-        result = cli_runner.invoke(
-            cli,
-            ["--from", "yaml", filename, "krishnamurti"])
+        result = cli_runner.invoke(cli, ["--from", "yaml", filename, "krishnamurti"])
         assert result.exit_code == 0
 
     import papis.yaml
+
     data = papis.yaml.yaml_to_data(filename)
 
     db = papis.database.get()
-    doc, = db.query_dict({"author": "Russell"})
+    (doc,) = db.query_dict({"author": "Russell"})
 
     assert doc["doi"] == data["doi"]
 
 
-def test_update_bibtex_cli(tmp_library: TemporaryLibrary,
-                           resource_cache: ResourceCache,
-                           monkeypatch: pytest.MonkeyPatch) -> None:
+def test_update_bibtex_cli(
+    tmp_library: TemporaryLibrary,
+    resource_cache: ResourceCache,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
     from papis.commands.update import cli
+
     cli_runner = PapisRunner()
 
     filename = os.path.join(resource_cache.cachedir, "update", "wannier.bib")
 
     import papis.utils
+
     with monkeypatch.context() as m:
-        m.setattr(papis.utils, "update_doc_from_data_interactively",
-                  update_doc_from_data_interactively)
+        m.setattr(
+            papis.utils,
+            "update_doc_from_data_interactively",
+            update_doc_from_data_interactively,
+        )
 
-        result = cli_runner.invoke(
-            cli,
-            ["--from", "bibtex", filename, "krishnamurti"])
+        result = cli_runner.invoke(cli, ["--from", "bibtex", filename, "krishnamurti"])
         assert result.exit_code == 0
 
     import papis.bibtex
-    data, = papis.bibtex.bibtex_to_dict(filename)
+
+    (data,) = papis.bibtex.bibtex_to_dict(filename)
 
     db = papis.database.get()
-    doc, = db.query_dict({"author": "Wannier"})
+    (doc,) = db.query_dict({"author": "Wannier"})
```

Comment:
Did black reformat this entire file? Haha. Not sure how those things are configured, but maybe there's a  knob to turn it off if there's no `[tool.black]` in `pyproject.toml` or something like that?

Having said that, we should probably try out a formatter, either `black` or `ruff format` at this point, so that things are vaguely consistent :disappointed:

Reply:
Yeah, it's `ruff` doing things. I'm also not quite sure how to configure stuff to get this working right. On the one hand, I do like to have a formatter as it makes sure I don't do silly things that mess with the linter. On the other hand, well these odd changes happen. Usually I turn it off when editing a line here or there in an existing doc, but keep it on when it's a new file (or I change most things as I did here). Not sure what's the best thing to do :thinking:

Reply:
> In this particular case, should I change at least these brackets back to what it was before?

No, it should be fine, since there's a couple of slightly different styles in the code anyway. We'll just need to add a formatter and do a cleanup.. maybe after 0.14 is released?

Reply:
Sounds good!